From 2e83bc545d3238fbcf55a4248c1f6f671b3824e3 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Mon, 29 Apr 2019 19:23:46 +0300
Subject: [PATCH 1/1] Check for a hash table size and nullable before lookups.

This should hopefully fix notifications under MATE 1.22. Tested under Solus.
---
 .vscode/settings.json     | 3 +++
 src/notification-window.c | 6 +++++-
 2 files changed, 8 insertions(+), 1 deletion(-)
 create mode 100644 .vscode/settings.json

diff --git a/.vscode/settings.json b/.vscode/settings.json
new file mode 100644
index 0000000..b5d37d7
--- /dev/null
+++ b/.vscode/settings.json
@@ -0,0 +1,3 @@
+{
+	"editor.detectIndentation": false
+}
\ No newline at end of file
diff --git a/src/notification-window.c b/src/notification-window.c
index 6e92f10..bb09192 100644
--- a/src/notification-window.c
+++ b/src/notification-window.c
@@ -298,13 +298,17 @@ void slate_notification_window_clear_actions(SlateNotificationWindow *self)
  */
 void slate_notification_window_set_hints(SlateNotificationWindow *self, GHashTable *hints)
 {
+    if (!hints && g_hash_table_size(hints) != 0) {
         GValue *hash = NULL;
 
         hash = g_hash_table_lookup(hints, "action-icons");
+    
         if (!hash || !G_VALUE_HOLDS_BOOLEAN(hash)) {
-                return;
+            return;
         }
+    
         self->action_icons = g_value_get_boolean(hash);
+    }
 }
 
 /*
-- 
2.21.0

